daemon/cluster/services.go

```diff
@@ -134,6 +138,35 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
+			switch serviceSpec.Task.GetGeneric().Kind {
+			case string(types.RuntimePlugin):
+				if s.TaskTemplate.PluginSpec == nil {
+					return errors.New("plugin spec must be set")
+				}
+				p := runtime.GenericPluginPayload{
+					Spec: *s.TaskTemplate.PluginSpec,
+				}
+				if encodedAuth != "" {
+					p.PullOptions = runtime.PluginPullOptions{RegistryAuth: encodedAuth}
```

Comment:
> I'm not sure I completely understand. There is no JSON version of the structure containing the registry auth.

Yes, that's exactly what I mean. Since this is not in the structure used by the JSON API, it is inherently not exposed (and inherently needs special handling for create/update).